### PR TITLE
add task 8

### PR DESCRIPTION
**Pull request by suzik86**

### CODE REVIEW

File: src/app/pages/add-course-page/add-course-page.component.ts [hunk 1]
```diff
@@ -19,23 +19,25 @@ export class AddCoursePageComponent {
...
-        this.isEdit = true;
-        const id = parseInt(activatedRouter.snapshot.paramMap.get('id'), 10);
-        this.course = this.coursesService.getItemById(id);
-      } else {
-        this.course = new Course();
-      }
-      localStorage.setItem('editableCourseName', this.course.title);
+  }
+
+  async ngOnInit(): Promise<void> {
```
Comment (katrin2404) [inline src/app/pages/add-course-page/add-course-page.component.ts]: Why do you use async here?

**Comment by katrin2404** [comment]
Rename AddCoursePageComponent, use CoursePageComponent, please.

File: src/app/pages/courses-page/components/courses-list/courses-list.component.ts [hunk 2]
```diff
@@ -8,30 +7,40 @@ import Course from '../course/course.types';
...
+  @Input() textFragment = '';
+  coursesList: Array<Course> = [];
 
   constructor(
-    private filterPipe: FilterPipe,
-    private coursesService: CoursesService,
-    ) {}
+    public coursesService: CoursesService,
+  ) {
+    this.coursesService.coursesChanged.subscribe(() => this.loadData());
```
Comment (katrin2404) [inline src/app/pages/courses-page/components/courses-list/courses-list.component.ts]: Don't use constructor for defining some value or something else. Best practice is to use constructor for DI. Define value for class properties in lifecycle hooks or declaration space.

File: src/app/pages/courses-page/components/courses-list/courses-list.component.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
-<ng-container *ngIf="filteredCourses.length; else noDataMessage">
+<ng-container *ngIf="coursesList.length; else noDataMessage">
   <ul>
-    <li *ngFor="let course of filteredCourses | orderBy: 'creationDate' : 'ASC'">
+    <li *ngFor="let course of coursesList">
```
Comment (katrin2404) [inline src/app/pages/courses-page/components/courses-list/courses-list.component.html:3]: Why did you remove orderBy pipe? Update this pipe if it's necessary.

File: src/app/pages/courses-page/courses-implementation.service.ts [hunk 4]
```diff
@@ -1,60 +1,41 @@
...
+                .set('textFragment', textFragment);
+
+    return await this.http.get<Course[]>('http://localhost:3004/courses', {params}).toPromise();
   }
 
-  createCourse(course): void {
-    course.id = this.courses.length + 1;
-    this.courses.push(course);
+  async createCourse(course): Promise<void> {
+    await this.http.post<void>('http://localhost:3004/courses/', course).toPromise();
```
Comment (katrin2404) [inline src/app/pages/courses-page/courses-implementation.service.ts]: Create constant for endpoint URL.

File: src/app/pages/courses-page/components/courses-list/courses-list.component.ts [hunk 5]
```diff
@@ -8,30 +7,40 @@ import Course from '../course/course.types';
...
-  ngDoCheck(): void {
-    this.courses = this.coursesService.list;
+  async ngOnInit(): Promise<void> {
+    await this.loadData();
   }
 
-  get coursesList(): Array<Course> {
-    return this.courses;
+  ngOnChanges(changes: SimpleChanges): void {
+    const item: SimpleChange = changes.item;
```
Comment (katrin2404) [inline src/app/pages/courses-page/components/courses-list/courses-list.component.ts]: Please review this component. You don't need so much loadData call and if you use subscribe you need to unsubscribe.

File: src/app/shared/components/header/header.component.html [hunk 6]
```diff
@@ -4,7 +4,7 @@
     <h1>VIDEO COURSE</h1>
   </a>
   <div class="user-info" *ngIf="authenticationService.isAuthenticated">
-    <span>{{authenticationService.currentUser.firstName}}</span>
+    <span>{{authenticationService.currentUser?.name.first}} {{authenticationService.currentUser?.name.last}}</span>
```
Comment (katrin2404) [inline src/app/shared/components/header/header.component.html]: Don't get data from service in template, use component for it. 
Also check this component I see errors in console.

File: src/app/pages/add-course-page/add-course-page.component.ts (hunk 1 above)
Comment (suzik86) [reply src/app/pages/add-course-page/add-course-page.component.ts]: I deleted it

File: src/app/pages/courses-page/components/courses-list/courses-list.component.ts (hunk 2 above)
Comment (suzik86) [reply src/app/pages/courses-page/components/courses-list/courses-list.component.ts]: I moved the subscription to the ngOnInit method.

File: src/app/pages/courses-page/components/courses-list/courses-list.component.html (hunk 3 above)
Comment (suzik86) [reply src/app/pages/courses-page/components/courses-list/courses-list.component.html:3]: Discussed and leave as it is for now.

File: src/app/pages/courses-page/courses-implementation.service.ts (hunk 4 above)
Comment (suzik86) [reply src/app/pages/courses-page/courses-implementation.service.ts]: Done

File: src/app/pages/courses-page/components/courses-list/courses-list.component.ts (hunk 5 above)
Comment (suzik86) [reply src/app/pages/courses-page/components/courses-list/courses-list.component.ts]: I added an unsubscribe from all subscriptions

File: src/app/shared/components/header/header.component.html (hunk 6 above)
Comment (suzik86) [reply src/app/shared/components/header/header.component.html]: Done
